MainMenuSample/Source/Game/ButtonFuctions.cs

```diff
@@ -0,0 +1,306 @@
+        public UIControl GraphicsSettingButton;
+        public UIControl AudioButton;
+        public UIControl OptionsBackButton;
+        public Actor OptionsMenuText;
+
+        public SceneReference GameScene;
+
+        //--- Private variables ---\\
+        private bool _audioEnabled;
+        private bool _highGrapicsPreset;
+        private bool _optionsMenuActive;
+        private bool _fullscreen;
+
+        private string _dataPath;
+        private string _directoryPath;
+        private string[] _lastSettingsReadValue;
+
+
+        public override void OnEnable()
+        {
+            PlayButton.Get<Button>().ButtonClicked += PlayButtonFunction;
+            OptionsMainMenuButton.Get<Button>().ButtonClicked += ToggleOptionsMenu;
+            OptionsBackButton.Get<Button>().ButtonClicked += ToggleOptionsMenu;
+            FullscreenButton.Get<Button>().ButtonClicked += ToggleFullscreen;
+            GraphicsSettingButton.Get<Button>().ButtonClicked += ModifyGraphicsSettingQuality;
+            AudioButton.Get<Button>().ButtonClicked += ToggleAudio;
+            QuitButton.Get<Button>().ButtonClicked += QuitButtonFunction;
+
+            // Generates the path to the settings file.
+            string fileName = SettingsFileName + "." + SettingsFileExtension;
+            
+            _dataPath = StringUtils.CombinePaths(Globals.ProductLocalFolder, fileName);
+            Debug.Log("Settings file location: " + _dataPath);
+
+            if (!File.Exists(_dataPath))
+            {
+                if (!Directory.Exists(Globals.ProductLocalFolder))
+                {
+                    Directory.CreateDirectory(Globals.ProductLocalFolder);
+                }
+                WriteSettingsFile(true, false, true);
+            }
+
+            _optionsMenuActive = true;
+            ToggleOptionsMenu(OptionsMainMenuButton.Get<Button>());
+
+            ReadSettingsFile();
+
+            string lastRead_Formed = "";
+            for (var readLine = 0; readLine < _lastSettingsReadValue.Length; readLine++)
+            {
+                lastRead_Formed = lastRead_Formed + Environment.NewLine + _lastSettingsReadValue[readLine];
+            }
+
+            bool gfxBool;
+            if (lastRead_Formed.Contains(GFX_Preset + " = " + SettingsFileTrueTerm + ";"))
+            {
+                gfxBool = true;
+            }else
+            {
+                gfxBool = false;
+            }
+
+            bool fullscreenBool;
+            if (lastRead_Formed.Contains(FullscreenPreset + " = " + SettingsFileTrueTerm + ";"))
+            {
+                fullscreenBool = true;
+            }else
+            {
+                fullscreenBool = false;
+            }
+
+            bool sfxBool;
+            if (lastRead_Formed.Contains(SFX_Preset + " = " + SettingsFileTrueTerm + ";"))
+            {
+                sfxBool = true;
+            }else
+            {
+                sfxBool = false;
+            }

```

Comment:
Personally I find that overall there are too many string concatenations and if statements like these throughout the source, this has the potential to be simplified perhaps atleast via ternary operator or use a different implementation entirely.

Reply:
You're right, I should've gotten something like: `bool sfxBool = lastRead_Formed.Contains(SFX_Preset + " = " + SettingsFileTrueTerm + ";")` and to have them be only 3 lines instead of 20 or so. Probably too late to change it.